src/sentry/integrations/github/client.py

```diff
@@ -43,17 +46,119 @@ def __repr__(self) -> str:
         return f"GithubRateLimit(limit={self.limit},rem={self.remaining},reset={self.reset})"
 
 
-class GitHubClientMixin(ApiClient):  # type: ignore
+class GithubProxyClient(IntegrationProxyClient):
+    def _get_installation_id(self) -> str:
+        """
+        Returns the Github App installation identifier.
+        This is necessary since Github and Github Enterprise integrations store the
+        identifier in different places on their database records.
+        """
+        return self.integration.external_id
+
+    @control_silo_function
+    def _refresh_access_token(self) -> str | None:
+        integration = Integration.objects.filter(id=self.integration.id).first()
+        if not integration:
+            return None
+
+        logger.info(
+            "token.refresh_start",
+            extra={
+                "old_expires_at": self.integration.metadata.get("expires_at"),
+                "integration_id": self.integration.id,
+            },
+        )
+        data = self.post(f"/app/installations/{self._get_installation_id()}/access_tokens")
```

Comment:
Do we need to handle networking errors here, or does the caller do that?

Reply:
I don't think we're handling network errors explicitly for most integrations -- I tried to avoid modifying any of the logic and just change the semantics to use the proxy client

Reply:
This may be something we could look into for the base client though.